presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/ExpressionConverter.java

```diff
@@ -174,6 +176,10 @@ private static Object getValue(Type type, Marker marker, ConnectorSession sessio
             return MILLISECONDS.toMicros((Long) marker.getValue());
         }
 
+        if (type instanceof DateType) {
```

Comment:
The newer version of Iceberg will allow Long->Date conversion, so conversion to int won't be necessary. Please see https://github.com/apache/incubator-iceberg/commit/eb9c75e41f1c0045595302710f2f6500f038e626.